### ISSUE TITLE
`nitro complete` instructions do not work on macOS

### ISSUE THREAD

**Issue by MattWilcox** [craftcms/docs#170]
### Description

Documented option for `nitro complete` on BigSur does not work.
https://craftcms.com/docs/nitro/2.x/commands.html#completion

```
echo "autoload -U compinit; compinit" >> ~/.zshrc

nitro completion zsh > "${fpath[1]}/_nitro"
zsh: no such file or directory: /usr/local/share/zsh/site-functions/_nitro
```

It's assuming directories exist when they do not?

- Nitro 2
- OSX 11.2.2

**Comment by mattstein** [comment]
Thanks for the issue @MattWilcox! You’re right; I was able to reproduce this and I’m wondering if you can confirm the fix.

1. Make sure `autoload -U compinit; compinit` appears—only once—in your `~/.zshrc`.
2. Run `mkdir -p "${fpath[1]}"` to safely ensure that your completions path (`/usr/local/share/zsh/site-functions`) exists.
3. Retry `nitro completion zsh > "${fpath[1]}/_nitro"`.
4. Start a new session or `source ~/.zshrc` to apply the autocompletion change.
5. Test completion with something like `nitro db import ~/Do` and tabbing to autocomplete a known path like `~/Downloads`. (Where `wnloads` popping in by itself means we’re winning.)

**Comment by MattWilcox** [comment]
```
mkdir -p "${fpath[1]}"
mkdir: /usr/local/share/zsh/site-functions: Permission denied
nitro completion zsh > "${fpath[1]}/_nitro" 
zsh: permission denied: /usr/local/share/zsh/site-functions/_nitro
```

First command works when sudo.
Second command fails even with sudo.

```
sudo nitro completion zsh > "${fpath[1]}/_nitro"
zsh: permission denied: /usr/local/share/zsh/site-functions/_nitro
```

**Comment by mattstein** [comment]
Well that’s not great. I’m not sure where completion directories *need* to live, though mine’s apparently `~/.zsh/completion` and I’m guessing this is why the permissions are fine for me.

Do you have other completions working in Zsh?

FWIW I’ve set up [p10k](https://github.com/romkatv/powerlevel10k) and [oh-my-zsh](https://ohmyz.sh/) on a fairly recent+fresh Big Sur install and my only `fpath` reference in `.zshrc` is

```
fpath=(~/.zsh/completion $fpath)
```

Trying to understand what setup we should expect and would love to know what yours looks like if you’re willing to share.

**Comment by MattWilcox** [comment]
I have not set up any completions, and other than adding some alias' the only other thing I've done with ZSH is installed the Spaceship prompt, so my .zshrc contains some Alias's and this:

```
# setup spaceship for the ZSH prompt; installed via npm... https://github.com/denysdovhan/spaceship-prompt
fpath=($fpath "/Users/mattwilcox/.zfunctions")
```

Otherwise it's essentially a blank file and stock BigSur.

**Comment by mattstein** [comment]
Thank you! I need to do some digging and experimenting but I’ll be back.

**Comment by mattstein** [comment]
Fix in Nitro [should be along shortly](https://github.com/craftcms/nitro/issues/295#issuecomment-800679439) so this will work as advertised!

@MattWilcox From testing, I’m fairly confident you need to set a Zsh completion path you have permissions to write and execute—so that setup snag may be a separate issue altogether. (There’s no single, recommended directory so we can’t be too prescriptive in the Nitro docs.)

**Comment by MattWilcox** [comment]
Thanks Matt; will take a look at that issue and see if it's just config weirdness. Looking forward to the release.

**Comment by mattstein** [comment]
FWIW I just merged an update for the [`completion`](https://craftcms.com/docs/nitro/2.x/commands.html#completion) command documentation that should be easier to follow. That combined with the Zsh fix in [Nitro 2.0.6](https://github.com/craftcms/nitro/releases/tag/2.0.6) should resolve this issue.

**Comment by MattWilcox** [comment]
Got that working, thanks... the permissions for the fpath folders are all root by default, and so too was the additional fpath folder created by the Spaceship Prompt installation. I changed the owners on the folder and was able to add _nitro at ~/.zfunctions

It's possible that assuming $fpath[1] is writable or predictable might be a little problematic.

**Comment by mattstein** [comment]
That’s weird @MattWilcox—I’m 80% sure I blindly followed instructions setting up Oh My Zsh and Powerlevel10k and my completion path is owned by `myuser:staff` and not `root`.

The underlying completion library [recommends this `$fpath[1]` approach](https://umarcor.github.io/cobra/#shell-completions), and you *might* be able to get away with adding `source <(nitro completion zsh)` to your profile but I need to figure out how to virtualize Big Sur and test a few times with a fresh OS before we make our own recommendations.

In the meantime, I’m glad you’ve got command completion working! 🎉
